src/core/Carousel/CarouselDot.tsx

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { useMergedRefs } from '../utils';
+
+type CarouselDotProps = {
+  /** Is this dot currently active? */
+  isActive?: boolean;
+  /** Should be set to true for dots that are one spot from the edge of truncation. The dot size becomes small.  */
+  isSmall?: boolean;
+  /** Should be set to true for dots that are at the edge of truncation. The dot size becomes even smaller.  */
+  isSmaller?: boolean;
+} & React.ComponentPropsWithoutRef<'button'>;
+
+/**
+ * `CarouselDot` is the actual "dot" component used to activate a slide on clicking.
+ * It should be used as a child of `CarouselDotsList`.
+ */
+export const CarouselDot = React.forwardRef<
+  HTMLButtonElement,
+  CarouselDotProps
+>((props, ref) => {
+  const { isActive, isSmaller, isSmall, className, ...rest } = props;
+
+  const justMounted = React.useRef(false);
+  const motionOk = React.useRef(
+    window.matchMedia?.('(prefers-reduced-motion: no-preference)').matches,
+  );
+
+  const dotRef = React.useCallback(
+    (el: HTMLButtonElement) => {
+      if (el && isActive) {
+        try {
+          el.scrollIntoView({
+            behavior:
+              !justMounted.current && motionOk.current ? 'smooth' : 'auto',
+            block: 'nearest',
+            inline: 'center',
+          });
+        } catch (_) {
+          // Safari fallback
+          el.parentElement?.scrollTo({
+            behavior:
+              !justMounted.current && motionOk.current ? 'smooth' : 'auto',
+            left: el.offsetLeft - el.parentElement?.offsetLeft,
+          });
+        }
+      }
+    },
+    [isActive],
+  );
+  const refs = useMergedRefs(ref, dotRef);
+
+  React.useEffect(() => {
+    if (justMounted.current) {
```

Comment:
Scroll down

Reply:
I did, cant see still
 
 
![image](https://user-images.githubusercontent.com/81580355/159906904-74400204-5b13-413c-8f41-a9cf6f92a04d.png)

Reply:
<img width="596" alt="Screen Shot 2022-03-24 at 7 32 04 AM" src="https://user-images.githubusercontent.com/9084735/159907569-7fe1f551-4c8c-4e3d-b50b-684063764bef.png">

Reply:
oh ok I see, there's a typo 😆 it should be initialized with `true`

Reply:
Ok fixed now.

To answer your question, it is needed for initializing default state with instant scroll. e.g. if the default dot is not in view, then we don't want it to smooth scroll when the page first loads.

This same logic is in CarouselSlider too.

Reply:
update: I've removed smooth scrolling from dots because it doesn't work in chrome using `scrollIntoView`, so all this code has been moved into `CarouselDotsList` and uses `scrollTo`. It's working in all browsers now